pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -706,15 +741,20 @@ func (r *ReconcileBareMetalHost) actionManageHostPower(prov provisioner.Provisio
 	// a delay.
 	result.RequeueAfter = time.Second * 60
 
-	if info.host.Status.PoweredOn == info.host.Spec.Online {
+	needsReboot := (!info.host.Status.PendingRebootSince.IsZero() &&
```

Comment:
I'm not sure how I feel about mixing the reboot logic with the standard power management logic. It limits us to rebooting after provisioning is done, for one thing, but it also complicates the logic here, which has to figure out if we're in the middle of a reboot operator, and if so where. I wonder if the logic of an actionReboot() function would be easier to follow because we would know the context is a reboot.